packages/laconia-event/src/apigateway/ApiGatewayInputHeaders.js

```diff
@@ -1,8 +1,18 @@
 module.exports = class ApiGatewayInputHeaders {
   constructor(eventHeaders) {
-    Object.assign(this, eventHeaders);
     Object.entries(eventHeaders).forEach(([headerName, headerValue]) => {
       this[headerName.toLowerCase()] = headerValue;
     });
+
+    const handler = {
+      get: (target, prop) => {
+        const isSymbol = typeof prop === "symbol";
+        const isExisting = prop in target;
+        const propertyKey = isSymbol || isExisting ? prop : prop.toLowerCase();
```

Comment:
What's the benefit of checking it like this, instead of doing `Object.prototype.hasOwnProperty.call(target, prop) ? prop.toLowerCase() : prop` ?

Reply:
Yes, happy for you to proceed.

The reason for the check above is that `hasOwnProperty()` does not check for inherited properties, hence things like `toString()` will stop working because the Proxy will change it to `tostring()` (from my rudimentary understanding of JS)